src/other/tables.jl

```diff
@@ -0,0 +1,32 @@
+using Tables, IteratorInterfaceExtensions
+
+Tables.schema(df::DataFrame) = NamedTuple{Tuple(names(df)), Tuple{eltypes(df)...}}
+Tables.AccessStyle(df::DataFrame) = Tables.ColumnAccess()
+Tables.istable(::Type{DataFrame}) = true
+Tables.rows(df::DataFrame) = eachrow(df)
+Tables.columns(df::DataFrame) = df
+
+function _isiterabletable(x::T) where {T}
+    Base.isiterable(T) || return false
+    return Base.IteratorEltype(x) === Base.HasEltype() ? eltype(x) <: NamedTuple : false
+end
+
+function DataFrame(x::T) where {T}
```

Comment:
I guess if we decide to add this catch-all definition we should also update docstring of `DataFrame` constructor explaining all the logic that happens here.